lua/toggleterm.lua

```diff
@@ -32,8 +33,12 @@ local persistent = {}
 ---   3. Default/base case perference size
 --- @param size number
 local function get_size(size)
-  local persist_size = preferences.direction == "horizontal" and persistent['height'] or persistent['width']
-  return (size and size > 0) and size or persist_size and persist_size or preferences.size
+  if preferences.persist_size then
```

Comment:
Think this should be
```suggestion
  if not preferences.persist_size then
```

Reply:
actually also just realised it shouldn't directly return `preferences.size` it should return the `size` passed in to the function so someone can still call `exec(23)` and have the specific size they required open

Reply:
Actually now that you mention that should it `return (size and size > 0) or preferences.size` so that it conforms to what was there before. Also yes will fix the not issue